state/binarystorage/binarystorage.go

```diff
@@ -46,11 +44,12 @@ func NewStorage(
 	}
 }
 
-func (s *toolsStorage) AddTools(r io.Reader, metadata Metadata) (resultErr error) {
-	// Add the tools tarball to storage.
-	path := toolsPath(metadata.Version, metadata.SHA256)
+// Add implements Storage.Add.
+func (s *binaryStorage) Add(r io.Reader, metadata Metadata) (resultErr error) {
+	// Add the binary file to storage.
+	path := fmt.Sprintf("binary/%s-%s", metadata.Version, metadata.SHA256)
```

Comment:
I think this should use the same path as before so we don't have to migrate the db.